app/models/adventure.rb

```diff
@@ -40,4 +53,9 @@ def editable_by?(current_user)
   def slug_source
     title
   end
+
+  def setting_password?
+    (password || password_confirmation) && has_password
```

Comment:
little rails nicety, you should be able to use the `has_password?` method, which calls `.presence` on the attribute value. In practice it's a straight alias for boolean fields, but conveys the boolean intent of the method call more directly